### PR TITLE
Enable dind for eventing-kafka repo

### PR DESCRIPTION
**Pull request by chaodaiG**
knative-sandbox/eventing-kafka repo needs DIND(docker in docker) for testing, enable this in all of its Prow jobs

/assign chizhg
/cc davyodom

### CODE REVIEW

**Comment by chaodaiG** [comment]
cc @davyodom

File: config/prow/config_knative.yaml [hunk 1]
```diff
@@ -435,12 +435,16 @@ presubmits:
   knative-sandbox/eventing-kafka:
     - build-tests: true
       dot-dev: true
+      needs-dind: true
     - unit-tests: true
       dot-dev: true
+      needs-dind: true
     - integration-tests: true
       dot-dev: true
+      needs-dind: true
```
Comment (chizhg) [inline config/prow/config_knative.yaml:444]: I assume only integration tests will need dind?

File: config/prow/config_knative.yaml [hunk 2]
```diff
@@ -1042,15 +1046,19 @@ periodics:
     - continuous: true
       go113: true
       dot-dev: true
+      needs-dind: true
     - nightly: true
       go113: true
       dot-dev: true
+      needs-dind: true
     - dot-release: true
```
Comment (chizhg) [inline config/prow/config_knative.yaml:1054]: The `dot-release` is missing `dot-dev: true`, maybe it was missed when it was added in the first time?

File: config/prow/config_knative.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (chaodaiG) [reply config/prow/config_knative.yaml:444]: nvm, found the answer: https://prow.knative.dev/view/gcs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka/2/pull-knative-sandbox-eventing-kafka-build-tests/1252404472584343552

File: config/prow/config_knative.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (chaodaiG) [reply config/prow/config_knative.yaml:1054]: probably was forgot, added

**Pull request merged**